src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -364,70 +298,83 @@
      * Checks whether the given stack is contained in this Inventory.
      * The stack size is ignored. Note this will return true if any amount
      * of the supplied stack is found. To check if an inventory contains at
-     * least an amount use {@link #contains(ItemStack)}.
+     * least a given quantity use {@link #contains(ItemStack)}.
      *
      * @param stack The stack to check for
+     *
      * @return True if the stack is present in this inventory
      */
     boolean containsAny(ItemStack stack);
 
+    // TODO remove from API? do we need to get a property relative to another parent in API?
     /**
-     * Returns the maximum size of any stack in this Inventory.
-     *
-     * @return The maximum stack size of this list
-     */
-    int getMaxStackSize();
-
-    /**
-     * Sets the maximum stack size of any stack in this ItemList.
-     *
-     * @param size The new maximum stack size
-     */
-    void setMaxStackSize(int size);
-
-    /**
-     * Gets the property with the default key defined in <em>this</em>
+     * Gets the property defined in <em>this</em>
      * inventory for the specified (immediate) sub-inventory.
      *
      * @param child The child inventory to inspect
      * @param property The property to retrieve the value for
      * @param <V> The property value type
+     *
      * @return The property value, if available
      */
     <V> Optional<V> getProperty(Inventory child, Property<V> property);
 
     /**
      * Gets a property defined directly on this Inventory if one is defined.
      * For sub-inventories this is effectively the same as
-     * <code>inv.getParent().getProperty(inv, property);</code> but for
+     * {@code inv.getParent().getProperty(inv, property);} but for
      * top-level inventories may include properties defined on the inventory
      * directly.
      *
      * @param property The property to retrieve the value for
      * @param <V> The property value type
+     *
      * @return The property value, if available
      */
     @Override
     <V> Optional<V> getProperty(Property<V> property);
 
     /**
-     * Query this inventory for inventories matching any of the supplied
-     * queries. Logical <code>OR</code> is applied between operands.
+     * Query this inventory with given {@link Query}
      *
-     * @param operations queries to check against
-     * @return the query result
+     * @param query The query
+     *
+     * @return The queried inventory
      */
-    Inventory query(QueryOperation<?>... operations);
+    Inventory query(Query query);
 
     /**
-     * Query this inventory for inventories matching
-     * the supplied {@link PropertyMatcher}.
+     * Query this inventory with given {@link QueryType} and {@code param}.
```

Comment:
Unsure about the OneParam, TwoParam system. jOOQ Does something similar in how they handle DatabaseRows and their types. This is inherently inflexible because you have to keep littering this interface with additional param methods and interfaces.

Just my thoughts.

Reply:
Alternative Ideas?